### DISCUSSION TITLE
Unable to Render Nextjs via Proxy

### DISCUSSION THREAD

**Discussion by vbuser2004** [coder/code-server#6708] [Q&A]
I'm new to code-server and love it! It works very well but I am unable to render NextJS application in development using the proxy. The only reference I have found about this is [this discussion](https://github.com/coder/code-server/discussions/5837) from 2022, but there was no fix or work-around. 

My set up is running code-server 4.21.2 and Debian 12 with a Caddy.

I am able to see the content but the images, css, and js files all show as 404 (it appears that the uri does not include the proxy details - /proxy/3000). 

I have tried to use re-writes & 'absproxy', but that has not solved the problem.

Is this a known issue with nextjs or is there something else I should look at in my set up? Appreciate any direction on how to solve this question. Thank you!

**Comment by code-asher** [comment]
If you can, use the subdomain-based proxy.  It will require either a wildcard certificate or at least the addition of `3000.your-domain.tld`.  A subdomain avoids all the weird issues of hosting at a sub-path.

But otherwise, I assume you already followed the instructions in https://github.com/coder/code-server/discussions/5837#discussioncomment-4316852?  If that is not working then it could be a bug in Next.js.

**Reply by vbuser2004** [reply]
Yes, I did try the suggestions in 5837. I'll try the subdomain route. TY!
